File: Niveau 3/Niveau 3.10/binaire.hs
--------------------------------------------------------------------------------
-- Name:        Affichage binaire
-- Purpose:     France-IOI niveau 3.10
--
-- Author:      Sébastien
--
-- Created:     16/10/2022
--------------------------------------------------------------------------------

main :: IO()
main = do
    n <- readLn :: IO Int
    putStrLn $ binaire n
    
binaire :: Int -> String
binaire n
    | n <= 1 = show n
    | otherwise = binaire (n `div` 2) ++ show (n `mod` 2)

File: Niveau 3/Niveau 3.10/binaire.py
#-------------------------------------------------------------------------------
# Name:        Affichage binaire
# Purpose:     France-IOI niveau 3.10
#
# Author:      Sébastien
#
# Created:     16/10/2022
#-------------------------------------------------------------------------------

def binaire(n):
    if n > 1:
        binaire(n // 2)
    print(n % 2, end='')

if __name__ == '__main__':
    n = int(input())
    binaire(n)
    print()
